src/client/state/settings.js

```diff
@@ -110,6 +111,15 @@ class Settings extends EventEmitter {
     return settings.isPeopleDrawer;
   }
 
+  getShowNotifications() {
+    if (typeof this.showNotifications === 'boolean') return this.showNotifications;
+
+    const settings = getSettings();
+    if (settings === null) return true;
+    if (typeof settings.showNotifications === 'undefined') return true;
```

Comment:
if user has not allow notification permission then setting showNotification to true is a bad idea